### ISSUE TITLE
False negative for SyntaxError::text

### ISSUE THREAD

**Issue by owillebo** [PyCQA/pylint#1432]
### Steps to reproduce
try:
  pass
except SyntaxError as ex:
  ex.text.rstrip()

### Current behavior
E:  4, 2: Class 'text' has no 'rstrip' member (no-member)

### Expected behavior
No issue

### pylint --version output
pylint 1.7.1,
astroid 1.5.2
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:42:59) [MSC v.1500 32 bit (Intel)]pylint

**Comment by rogalski** [comment]
Thanks for submitting an issue!

Some basic debug:

It seems like all non-standard exception instance attributes are inferred as dummy `ClassDef`s and not actual types.

Given:

```
import astroid

code = """
try:
  pass
except SyntaxError as ex:
  ex #@
  ex.args #@
  ex.text #@
  ex.lineno #@
"""

for node in astroid.extract_node(code):
    print(node.as_string(), *node.infer())
```

We receive:
```
ex Instance of builtins.SyntaxError
ex.args Tuple.tuple(ctx=None,
            elts=(<Const.str l.None at 0x10f16e470>,))
ex.text ClassDef.text(name='text',
              doc='exception text',
              decorators=None,
              bases=[],
              body=[])
ex.lineno ClassDef.lineno(name='lineno',
                doc='exception lineno',
                decorators=None,
                bases=[],
                body=[])
```

Our [`interpreter.objectmodel.ExceptionInstanceModel`](https://github.com/PyCQA/astroid/blob/master/astroid/interpreter/objectmodel.py#L551) define some extra logic for `.args` and `.__traceback__` which exists for all BaseInstance subclasses, but I'm not sure how class-specific attributes (like those in this issue) should be handled.
